src/target/source/codegen_opencl.cc

```diff
@@ -89,6 +89,17 @@ void CodeGenOpenCL::InitFuncState(const PrimFunc& f) {
 
 void CodeGenOpenCL::PrintFuncPrefix() { stream << "__kernel void"; }
 
+void CodeGenOpenCL::PreFunctionBody(const PrimFunc& f) {
+  for (Var arg : f->params) {
+    auto ptr_type = arg->type_annotation.as<PointerTypeNode>();
+    if (ptr_type && runtime::IsTextureStorage(std::string(ptr_type->storage_scope))) {
+      this->stream << "  const sampler_t image_sampler = "
```

Comment:
nit: Will we create this variable before function body in global scope? 
If yes, then why we need this indent before `const sampler_t ...`?
I suppose it should be something like that:
```
const sampler_t image_sampler = "CLK_NORMALIZED_COORDS_FALSE | CLK_ADDRESS_CLAMP | CLK_FILTER_NEAREST;\n";
__kernel void kernel_name(..) {
    // ...
}
```
Am I right?

Reply:
Its the first line of function body.

`__kernel void kernel_name(..) {`
`      const sampler_t image_sampler = "CLK_NORMALIZED_COORDS_FALSE | CLK_ADDRESS_CLAMP | CLK_FILTER_NEAREST;`
`    // ...`
`}`

Reply:
Ok, thank you for the clarification! LGTM.